compiler-plugin/src/test/kotlin/arrow/meta/plugins/optics/DSLTests.kt

```diff
@@ -0,0 +1,21 @@
+package arrow.meta.plugins.optics
+
+import org.junit.jupiter.api.Test
+
+class DSLTests {
+
+  @Test
+  fun `DSL is generated for complex model`() {
+    """
+      |$imports
+      |$dslModel
+      |val john = Employee("Audrey Tang",
+      |       Company("Arrow",
+      |               Address("Functional city",
+      |                       Street(42, "lambda street"))))
+      |val modify = Employee.company.address.street.name.modify(john, String::toUpperCase)
+      |val r = modify.company?.address?.street?.name
```

Comment:
```suggestion
      |val r = modify.company.address.street.name
```

These are not nullable in `dslModel`, we can make them `nullable` since that should also be supported. Those should then be generated as `Optional`, and still, be composable.

Reply:
In this case, I'm accessing an instance of `john` which is poorly named as `modify`